src/position.rs

```diff
@@ -178,11 +178,20 @@ fn lsp_range_to_kakoune_utf_8_code_units(range: &Range) -> KakouneRange {
 
 fn kakoune_position_to_lsp_utf_8_code_points(position: &KakounePosition, text: &Rope) -> Position {
     // -1 because LSP & Rope ranges are 0-based, but Kakoune's are 1-based.
-    let line = position.line - 1;
-    let character = text
-        .line(line as _)
-        .byte_to_char((position.column - 1) as _) as _;
-    Position { line, character }
+    let line_idx = position.line - 1;
+    if line_idx as usize >= text.len_lines() {
+        return Position { line: line_idx, character: position.column - 1 };
+    }
+
+    let line = text.line(line_idx as _);
+
+    if (position.column - 1) as usize >= line.len_bytes() {
+        return Position { line: line_idx, character: position.column - 1 };
```

Comment:
I'm having similar problems, could it be a timing issue where the language server works with an outdated document? The coordinates are out of range, I think that also happens when only using ASCII characters.

The two early returns are edge cases where we may get incorrect results. Does it make sense to log a warning here? Because normally this case should not happen.

I would extract `position.column - 1` to a variable.

Reply:
> I'm having similar problems, could it be a timing issue where the language server works with an outdated document? The coordinates are out of range, I think that also happens when only using ASCII characters.

As mentioned in the PR description, I am pretty certain it is this. I think this can happen with non-ASCII too, but I don't have much unicode in my document.

> The two early returns are edge cases where we may get incorrect results. Does it make sense to log a warning here? Because normally this case should not happen.

No, because it's not really an "edge case" per se; it can happen very trivially by just pressing backspace while the LSP is thinking; logging a warning would be way too verbose. IMO this function should've probably returned a `Result`, but ensuring those make it further upstream is a lot of effort, relatively

> I would extract `position.column - 1` to a variable.

Reasonable; this is pretty much the patch i wrote locally when I got annoyed at it repeatedly breaking, and I wanted to at least get some people's comments on this problem :)